tasks/main.yml

```diff
@@ -1,9 +1,14 @@
 # SPDX-License-Identifier: BSD-3-Clause
 # get service facts, used in defaults/main.yml
 ---
-- name: Check which services are running
-  service_facts:
-  no_log: true
+# If NetworkManager is running, assume that 'nm' is currently in-use,
+# otherwise initscripts
+- name: Get current provider
+  shell: |
+    for f in /proc/*/exe;
+      do [[ "$(readlink -f "${f}")" == /usr/sbin/NetworkManager ]] && echo nm && break
+    done || echo -n initscripts
```

Comment:
hmm - what about this:
```
command: systemctl is-enabled NetworkManager.service
```
that will tell you if the service is enabled.  If you want to find out if it is active:
```
command: systemctl is-active NetworkManager.service
```
man systemctl
```
       is-active PATTERN...
           Check whether any of the specified units are active (i.e. running).
           Returns an exit code 0 if at least one is active, or non-zero
           otherwise. Unless --quiet is specified, this will also print the
           current unit state to standard output.
```

Reply:
This was my initial idea, too, however there is no systemctl on CentOS 6, so it would require more checks...

Reply:
Also I tried to avoid too many extra tools to make it work in containers, too. Currently, it only depends on coreutils.

Reply:
Note that the approach in timesync also uses `systemctl` or `service` ... is it causing real problems? How would starting and enabling the service work in a container that does not have `systemctl` or `service`?

Reply:
Using systemctl means more complexity:

1. If systemctl is not installed -> initscripts
2. If systemctl is installeed
2.1. If NetworkManager is running -> nm
2.2 If NetworkManager is not running -> initscripts

The note about avoiding extra dependencies is why I chose that approach instead of using `ps` or `grep` etc.

However, it would be great if you describe the problems with the current approach, so we can find a solution for the problem instead of discussing alternative solutions without a clear explanation what problem they solve that the proposed solution does not.

Reply:
Are you actually worried about NetworkManager running on EL 6? From 1. it seems that you are not and if this is the case, it simplifies things, because if `systemctl` is not installed, it will fail, just as in case `systemctl is-active` is invoked on newer systems with NetworkManager not running.

Reply:
> Are you actually worried about NetworkManager running on EL 6? From 1. it seems that you are not and if this is the case, it simplifies things, because if `systemctl` is not installed, it will fail, just as in case `systemctl is-active` is invoked on newer systems with NetworkManager not running.

No, I don't care about NetworkManager running on EL 6. What is the full pattern with `systemctl is-active` that you propose, maybe this will clarify things. For, the problem with 1. was that it would fail the command module while 2.2 would report a different return code. But it seems that a different return code fails the module as well.

Something like this seems to work, too. Do you like this better? Do you prefer a construct with failing module that is ignored and then the failure is checked?
```
ansible -i rhel6, -m shell -a "systemctl is-enabled NetworkManager.service &>/dev/null  && echo nm || echo initscripts" all
```